cli/tests/test_working_copy.rs

```diff
@@ -30,33 +30,57 @@ fn test_snapshot_large_file() {
         This will increase the maximum file size allowed for new files, for this command only.
-    ");
+    "#);
 
     // test with a larger file using 'KB' human-readable syntax
     test_env.add_config(r#"snapshot.max-new-file-size = "10KB""#);
     let big_string = vec![0; 1024 * 11];
-    std::fs::write(repo_path.join("large"), big_string).unwrap();
+    std::fs::write(repo_path.join("large"), &big_string).unwrap();
     let (stdout, stderr) = test_env.jj_cmd_ok(&repo_path, &["file", "list"]);
     insta::assert_snapshot!(stdout, @"empty");
-    insta::assert_snapshot!(stderr, @r"
+    insta::assert_snapshot!(stderr, @r#"
     Warning: Refused to snapshot some files:
       large: 11.0KiB (11264 bytes); the maximum size allowed is 10.0KiB (10240 bytes)
     Hint: This is to prevent large files from being added by accident. You can fix this by:
       - Adding the file to `.gitignore`
       - Run `jj config set --repo snapshot.max-new-file-size 11264`
         This will increase the maximum file size allowed for new files, in this repository only.
-      - Run `jj --config snapshot.max-new-file-size=11264 st`
+      - Run `jj --config snapshot.max-new-file-size=11264 st
         This will increase the maximum file size allowed for new files, for this command only.
-    ");
+    "#);
+
+    // test with file track for hint formatting
+    std::fs::write(repo_path.join("large2"), big_string).unwrap();
+    let (stdout, stderr) = test_env.jj_cmd_ok(&repo_path, &["file", "track", "large", "large2"]);
+    insta::assert_snapshot!(stdout, @"");
+    insta::assert_snapshot!(stderr, @r#"
+    Warning: Refused to snapshot some files:
+      large: 11.0KiB (11264 bytes); the maximum size allowed is 10.0KiB (10240 bytes)
+      large2: 11.0KiB (11264 bytes); the maximum size allowed is 10.0KiB (10240 bytes)
+    Hint: This is to prevent large files from being added by accident. You can fix this by:
+      - Adding the file to `.gitignore`
+      - Run `jj config set --repo snapshot.max-new-file-size 11264`
+        This will increase the maximum file size allowed for new files, in this repository only.
+      - Run `jj --config snapshot.max-new-file-size=11264 st
+        This will increase the maximum file size allowed for new files, for this command only.
+    Warning: Refused to snapshot some files:
+      large: 11.0KiB (11264 bytes); the maximum size allowed is 10.0KiB (10240 bytes)
+      large2: 11.0KiB (11264 bytes); the maximum size allowed is 10.0KiB (10240 bytes)
+    Hint: This is to prevent large files from being added by accident. You can fix this by:
+      - Run `jj config set --repo snapshot.max-new-file-size 11264`
+        This will increase the maximum file size allowed for new files, in this repository only.
+      - Run `jj --config snapshot.max-new-file-size=11264 file track large large2`
+        This will increase the maximum file size allowed for new files, for this command only.
```

Comment:
This shows a weird edge-case: calling `file track` on a large file when it's not needed (i.e., the file is in the `auto-track` fileset and would've been tracked anyways) will result in a double output like that, one for the "auto snapshotting" which occurs for all commands, and one specific to `file track`

Reply:
Seems fine to leave as is for now. Maybe `track.rs` should subtract `auto_tracking_matcher()` from the matcher it passes into `snapshot_options_with_start_tracking_matcher()`